impress-org/givewp
# Fix PayPal Donation webhooks in live mode

Resolves #5402 

## Description

So a while back I found that there was an error in the [PayPal Event signature verification docs](https://developer.paypal.com/docs/api/webhooks/v1/#verify-webhook-signature-post-request-body). It says to grab the values from the headers. The `auth_algo` parameter, for example, comes from the `PAYPAL-AUTH-ALGO` header in the event request. I pointed out that the header is not actually in all caps, it's in studly case: `Paypal-Auth-Algo`.

I turns out I was half right. The headers are in studly case _in sandbox mode_, but apparently they are in all caps _in live mode_. In other words, the header shape sent by PayPal is inconsistent. This is not ideal, but it is the way it is.

This PR checks for both shapes to handle either situation.

## Affects

PayPal Donations webhook event handling

## Pre-review Checklist


## Testing Instructions

These steps must be done on a site that's reachable by PayPal (not local).

1. Make a one-time donation
2. Pull up the transaction for the donation within PayPal
3. Refund the transaction
4. Wait a few minutes to give the webhook time
5. Check the donation in GiveWP — it should be marked as refunded
